stage1/init/common/pod.go

```diff
@@ -412,48 +413,27 @@ func parseUserGroup(p *stage1commontypes.Pod, ra *schema.RuntimeApp) (int, int,
 // It returns the actual path relative to the app rootfs for the given path.
 // This is needed for absolute symlinks - we are in a different rootfs.
 func EvaluateSymlinksInsideApp(appRootfs, path string) (string, error) {
-	link := appRootfs
-
-	paths := strings.Split(path, "/")
-	for i, p := range paths {
-		next := filepath.Join(link, p)
-
-		if !strings.HasPrefix(next, appRootfs) {
-			return "", fmt.Errorf("path escapes app's root: %q", path)
-		}
-
-		fi, err := os.Lstat(next)
-		if err != nil {
-			if os.IsNotExist(err) {
-				link = filepath.Join(append([]string{link}, paths[i:]...)...)
-				break
-			}
-			return "", err
-		}
-
-		if fi.Mode()&os.ModeType != os.ModeSymlink {
-			link = filepath.Join(link, p)
-			continue
-		}
+	chroot, err := newChroot(appRootfs)
```

Comment:
does it make sense to `defer chroot.Escape()` here? So our root is where we want it to be if we error out?